File: Data Structures & Algorithms/sliding-window-maximum/submission-1.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        int curMax = 0;
        deque<int> q;
        int l = 0;
        vector<int> answer(nums.size()- k +1,0);
        for(int r = 0; r < nums.size();r++)
        {
            while(!q.empty() && nums[r] > nums[q.back()])
                q.pop_back();
            q.push_back(r);
            if(l > q.front())
                q.pop_front();
            if((r+1) >=k)
            {
                answer[l] = nums[q.front()];
                l++;
            }
            
        }
    return answer;
        
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        answer = []
        for i in range(len(nums)):
            cur = 1
            for j in range(len(nums)):
                if j == i :
                    continue
                cur *= nums[j]
            answer.append(cur)
        return answer

                

        